app/editor/src/hooks/api-editor/editor/useApiStorage.ts

```diff
@@ -0,0 +1,118 @@
+import { AxiosResponse } from 'axios';
+import React from 'react';
+import { defaultEnvelope, ILifecycleToasts, toQueryString } from 'tno-core';
+import { extractFileName } from 'utils';
+
+import { IFolderModel, IItemModel, useApi } from '..';
+
+/**
+ * Common hook to make requests to the API.
+ * @returns CustomAxios object setup for the API.
+ */
+export const useApiStorage = (
+  options: {
+    lifecycleToasts?: ILifecycleToasts;
+    selector?: Function;
+    envelope?: typeof defaultEnvelope;
+    baseURL?: string;
+  } = {},
+) => {
+  const api = useApi(options);
+
+  return React.useRef({
+    getFolder: (path?: string, location?: string) => {
+      const params = {
+        path,
+      };
+      return api.get<IFolderModel, AxiosResponse<IFolderModel, never>, any>(
+        `/editor/storage${location ? `/${location}` : ''}?${toQueryString(params)}`,
+      );
+    },
+    upload: (
+      path: string,
+      file: File,
+      overwrite?: boolean,
+      location?: string,
+      onUploadProgress?: (progressEvent: any) => void,
+    ) => {
+      const params = {
+        path,
+        overwrite,
+      };
+      const formData = new FormData();
+      formData.append('files', file, file.name);
+      return api.post<IItemModel, AxiosResponse<IItemModel, never>, any>(
+        `/editor/storage${location ? `/${location}` : ''}/upload?${toQueryString(params)}`,
+        formData,
+        { onUploadProgress },
+      );
+    },
+    stream: async (
+      path: string,
+      location?: string,
+      onUploadProgress?: (progressEvent: any) => void,
+    ) => {
+      const params = {
+        path,
+      };
+      var response = await api.get<any, AxiosResponse<any, never>, any>(
+        `/editor/storage${location ? `/${location}` : ''}/stream?${toQueryString(params)}`,
+        {
+          responseType: 'stream',
+          headers: { accept: '*.*' },
+          onUploadProgress,
+        },
+      );
+
+      const uri = window.URL.createObjectURL(new Blob([response.data]));
+      const link = document.createElement('a');
+      link.href = uri;
+      link.setAttribute('download', new Date().toDateString());
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+      return response;
+    },
+    download: async (path: string, fileName?: string, location?: string) => {
+      const params = {
+        path,
+      };
+      var response = await api.get<any, AxiosResponse<any, never>, any>(
+        `/editor/storage${location ? `/${location}` : ''}/download?${toQueryString(params)}`,
+        {
+          responseType: 'blob',
+          headers: { accept: '*.*' },
+        },
+      );
+
+      const uri = window.URL.createObjectURL(new Blob([response.data]));
+      const link = document.createElement('a');
```

Comment:
I'd like to make an option that allows the user more control when downloading files.  Presently this auto starts the download to their default/last location.